securesystemslib/metadata.py

```diff
@@ -97,3 +101,87 @@ def pae(self) -> bytes:
             len(self.payload),
             self.payload,
         )
+
+    def sign(self, signer: Signer) -> Signature:
+        """Sign the payload and create the signature.
+
+        Arguments:
+            signer: A "Signer" class instance.
+
+        Raises:
+            TypeError: If "signer" is not a instance of "Signer" class.
+
+        Returns:
+            A "Signature" instance.
+        """
+
+        if not isinstance(signer, Signer):
+            raise TypeError(f"expected type 'Signer', got {type(signer).__name__}")
+
+        signature = signer.sign(self.pae)
+        self.signatures.append(signature)
+
+        return signature
+
+    def verify(self, keys: KeyList, threshold: int) -> List[str]:
+        """Verify the payload with the provided verifier.
+
+        Arguments:
+            keys: A list of a "str" and "Key" class instance.
+            threshold: Number of signatures needed to pass the verification.
+
+        Raises:
+            TypeError: If key in "keys" is not a instance of "Key" class.
+            ValueError: If "threshold" is not valid.
+            SignatureVerificationError: If "recognized_signers" are less than provided
+                threshold.
+
+        Returns:
+            recognized_signers: list of key names for which verification succeeds.
+        """
+
+        recognized_signers = []
+        used_keyids = []
+        pae = self.pae
+
+        # checks for threshold value
+        if len(keys) < threshold or threshold <= 0:
+            raise ValueError(f"Threshold must be between 0 and {len(keys)}")
+
+        for signature in self.signatures:
+            for (name, key) in keys:
+                if not isinstance(key, Key):
+                    raise TypeError(f"expected type 'Key', got {type(key).__name__}")
+
+                # If key and signature include keyIDs but do not match skip.
+                if (
+                    signature.keyid is not None
+                    and key.keyid is not None
+                    and signature.keyid != key.keyid
+                ):
+                    continue
+
+                # If a key verifies the signature, we exit and use the result.
+                if key.verify(signature, pae):
+                    # from go-securesystemslib, keyid only counted once
```

Comment:
I don't see how we check if a key comes from `go-securesystemslib` as it sounds that specifically in the case of `go-securesystemslib` we count them once.